assets/js/components/notifications/dashboard-core-site-alerts.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * DashboardCoreSiteAlerts component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Notification from 'GoogleComponents/notifications/notification';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getNotifications,
+	acceptNotification,
+	dismissNotification,
+} from './site';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+class DashboardCoreSiteAlerts extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			notifications: false,
+		};
+	}
+
+	async componentDidMount() {
+		try {
+			const notifications = await getNotifications();
+			this.setState( { notifications } );
+		} catch ( error ) {
+			// eslint-disable-next-line no-console
+			console.warn( 'Error caught while fetching notifications', error );
+		}
+	}
+
+	render() {
+		const { notifications } = this.state;
+
+		if ( ! Array.isArray( notifications ) ) {
+			return null;
+		}
+
+		return notifications.map(
+			( notification ) => (
+				<Notification
+					key={ notification.id }
+					id={ notification.id }
+					title={ notification.title || '' }
+					description={ notification.content || '' }
+					learnMoreURL={ notification.learnMoreURL || '' }
+					learnMoreLabel={ notification.learnMoreLabel || '' }
+					ctaLink={ notification.ctaURL || '' }
+					ctaLabel={ notification.ctaLabel || '' }
+					ctaTarget={ notification.ctaTarget || '' }
+					dismiss={ notification.dismissLabel || __( 'OK, Got it!', 'google-site-kit' ) }
+					isDismissable={ notification.dismissible }
+					onCTAClick={ async () => {
+						await acceptNotification( notification.id );
```

Comment:
We need to make the notice dismiss here, also when the user accepts it. It could either be through manual logic in this callback here, or alternatively we could introduce some kind of `dismissOnCTA` flag prop in `Notification` to generally support that?

Reply:
The thing is right now, it opens the link in a new tab (which is what should happen). The problem is then the notice still remains open in the other window. We need to make sure it dismissed on the JS side.

Reply:
I just pushed a small update to address this although it needs testing still.

Basically, if no `onCTAClick` handler is passed, the notification will dismiss on CTA click if `dismissible`.

If an `onCTAClick` handler _is passed_ then that callback is responsible for calling `dismiss` or not, which is passed to the callback as the second argument (first being the event). If the event isn't dismissible, the `dismiss` function is a no-op.